src/neptune/new/attributes/series/series.py

```diff
@@ -25,34 +27,55 @@
 )
 
 from neptune.new.attributes.attribute import Attribute
-from neptune.new.internal.operation import Operation
+from neptune.new.internal.operation import LogOperation
 from neptune.new.internal.utils import (
     is_collection,
+    verify_collection_type,
     verify_type,
 )
+from neptune.new.internal.utils.iteration import get_batches
 from neptune.new.types.series.series import Series as SeriesVal
 
-Val = TypeVar("Val", bound=SeriesVal)
-Data = TypeVar("Data")
+ValTV = TypeVar("ValTV", bound=SeriesVal)
+DataTV = TypeVar("DataTV")
+LogOperationTV = TypeVar("LogOperationTV", bound=LogOperation)
 
 
-class Series(Attribute, Generic[Val, Data]):
+class Series(Attribute, Generic[ValTV, DataTV, LogOperationTV]):
+    MAX_BATCH_SIZE = None
+    operation_cls: type(LogOperationTV) = None
```

Comment:
> Isn't it some bad practice?
I don't know, I think it's common practice in Django, when you override values like `template_name`, `response_class` etc.

>  Does access to these variables using self. even work?
Yes that's what Python does by default:
1. Tries to to obtain instance variable stored in `slots` or internal `__dict__`
2. falls back to class attributes

Reply:
Doesn't abstract or class method sound like a better idea for you? You don't even have default values here and nothing forces you to override them in derived class.

Reply:
I get your point, but what you're suggesting looks like Java in Python and I don't feel it.

Maybe I just used to fucked up Python patterns, because to me even overriding `__init_sublcass__` like [here](https://stackoverflow.com/a/64397790/1565454) looks like a solution in this case...

I don't know. Maybe @Raalsky has some opinion on writing Python.

Reply:
I may miss something but the class properties are pretty useful things like the ground for descriptors and dataclasses fields. 

> Doesn't abstract or class method sound like a better idea for you?
> Doesn't abstract or class method sound like a better idea for you? You don't even have default values here and nothing forces you to override them in derived class.
You can as well override any abstract method, class method etc. (I think so :D).

I don't wanna dig deeply into this PR but it looks ok as a first impression.

Reply:
@Raalsky I agree with class properties being pretty useful in general but here they are used as something to be overriden in subclasses but python does not force you to do it and they don't have any reasonable default. If you add new subclass in future would it be obvious for you to override them?

Reply:
> here they are used as something to be overriden in subclasses but python does not force you to do it and they don't have any reasonable default. If you add new subclass in future would it be obvious for you to override them?

@aniezurawski 

Are you ok with `__init_sublcass__` as solution for your problem?

Reply:
I never heard of it earlier so I have no intuition about possible use cases and best practices but it looks like a good idea.